### DISCUSSION TITLE
New setup on a raspbery pi3

### DISCUSSION THREAD

**Discussion by sma0071** [nathanmarlor/foxess_modbus#658] [Support] [answered]
Hi, 
I had my setting working fine before on a VM. I have had to move to a pi3 as my computer died. 
I have followed the guide and have installed HACS, I installed Fox ESS modbus though HACS. I am getting the below error:

Logger: homeassistant.config_entries
Source: config_entries.py:604
First occurred: 4:36:09 PM (2 occurrences)
Last logged: 4:38:32 PM

Error setting up entry FoxESS - Modbus for foxess_modbus
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 604, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/foxess_modbus/__init__.py", line 116, in async_setup_entry
    create_controller(client, inverter)
  File "/config/custom_components/foxess_modbus/__init__.py", line 75, in create_controller
    inverter_connection_type_profile_from_config(inverter),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/foxess_modbus/inverter_profiles.py", line 371, in inverter_connection_type_profile_from_config
    return INVERTER_PROFILES[inverter_config[INVERTER_BASE]].connection_types[inverter_config[INVERTER_CONN]]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'LAN'

**Comment by canton7** [comment]
What inverter type do you have? Did you configure it after installing foxess_modbus, or did you restore the config from your previous setup?

**Reply by sma0071** [reply]
I have a H1-6.0-E-G2. It a completly fresh install, I lost all my configuration from my old server. it used to work perfectly.

**Reply by canton7** [reply, marked as answer]
I'm guessing you clicked "Direct connection to inverter" in the setup wizard, rather then selecting the correct modbus adapter type?

**Reply by canton7** [reply]
Hmm, we really shouldn't crash if someone does that! I'll have to have a think about the best way to handle that

**Reply by canton7** [reply]
Tracked by #659

**Comment by sma0071** [comment]
Ignore my problem. Its all sorted I used the wrong setup option when configuring the EW11.
